toolchain/ltrace.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import git
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+backend = SoftwareManager()
+dist = distro.detect()
+
+
+class Ltrace(Test):
+
+    def setUp(self):
+        """
+        Install  all the required dependencies
+        Building source tarball requires packages specific to os
+        that needs to be installed, if not installed test will stop
+        a library function check_repo() should be implemented.
+        """
+
+        os_deps = ['git', 'wget', 'rpm', 'autoconf',
+                   'automake', 'dejagnu', 'binutils']
+
+        if dist.name == 'sles':
+            sles_deps = ['build', 'libdw-devel', 'libelf-devel',
+                         'elfutils', 'binutils-devel', 'libtool', 'gcc-c++']
+            pkgs = os_deps + sles_deps
+
+        elif dist.name == 'redhat':
+            rhel_deps = ['elfutils-devel', 'elfutils-libelf-devel',
+                         'elfutils-libelf', 'elfutils-libs', 'libtool-ltdl']
+            pkgs = os_deps + rhel_deps
+
+        elif dist.name == 'ubuntu':
+            ubuntu_deps = ['elfutils', 'libelf-dev', 'libtool',
+                           'libelf1', 'librpmbuild3', 'binutils-dev']
+            pkgs = os_deps + ubuntu_deps
+        else:
+            print "Unsupported OS!"
+
+        i = 0
+        while i < len(pkgs):
+            if backend.install(pkgs[i]):
+                print "%s installed successfull" % pkgs[i]
+            else:
+                print "Error: Check repository !"
+                exit(1)
```

Comment:
Please never use `exit` directly. It works, but is not really nice. The most suitable would be `self.skip`, which says the test is not available on your machine. I'd also agree with `self.error` which says unrelated error occurred.